internal/command/container.go

```diff
@@ -0,0 +1,393 @@
+package command
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fyne-io/fyne-cross/internal/icon"
+	"github.com/fyne-io/fyne-cross/internal/log"
+	"github.com/fyne-io/fyne-cross/internal/volume"
+	"golang.org/x/sys/execabs"
+)
+
+const (
+	// fyneBin is the path of the fyne binary into the docker image
+	fyneBin = "/usr/local/bin/fyne"
+	// gowindresBin is the path of the gowindres binary into the docker image
+	gowindresBin = "/usr/local/bin/gowindres"
+)
+
+type ContainerRunner interface {
+	NewImageContainer(arch Architecture, OS string, image string) ContainerImage
+
+	Debug(v ...interface{})
+	GetDebug() bool
+}
+
+type AllContainerRunner struct {
+	ContainerRunner
+
+	Env  map[string]string // Env is the list of custom env variable to set. Specified as "KEY=VALUE"
+	Tags []string          // Tags defines the tags to use
+
+	vol volume.Volume
+
+	debug bool
+}
+
+type ContainerImage interface {
+	Cmd(vol volume.Volume, opts Options, cmdArgs []string) *execabs.Cmd
+	Run(vol volume.Volume, opts Options, cmdArgs []string) error
+	Prepare() error
+	Finalize(srcFile string, packageName string) error
+
+	GetArchitecture() Architecture
+	GetOS() string
+	GetID() string
+	GetTarget() string
+	GetEnv(string) (string, bool)
+	SetEnv(string, string)
+	SetMount(string, string)
+	AppendTag(string)
+
+	GetRunner() ContainerRunner
+}
+
+type ContainerMountPoint struct {
+	LocalHost   string
+	InContainer string
+}
+
+type AllContainerImage struct {
+	Architecture        // Arch defines the target architecture
+	OS           string // OS defines the target OS
+	ID           string // ID is the context ID
+
+	Env   map[string]string     // Env is the list of custom env variable to set. Specified as "KEY=VALUE"
+	Tags  []string              // Tags defines the tags to use
+	Mount []ContainerMountPoint // Mount point between local host [key] and in container point [target]
+
+	DockerImage string // DockerImage defines the docker image used to build
+}
+
+func NewContainerRunner(context Context) ContainerRunner {
+	if context.Engine.IsDocker() || context.Engine.IsPodman() {
+		return NewLocalContainerRunner(context)
+	}
+	return nil
+}
+
+func (a *AllContainerRunner) Debug(v ...interface{}) {
+	if a.debug {
+		log.Debug(v...)
+	}
+}
+
+func (a *AllContainerRunner) GetDebug() bool {
```

Comment:
I see why `Get` is needed here because you have mixed the noun and verb of `Debug`, is this perhaps `DebugEnabled` or even `Debugging`?